Modules/Core/Common/include/itkSphereSpatialFunction.hxx

```diff
@@ -60,7 +60,7 @@ SphereSpatialFunction<VImageDimension, TInput>::PrintSelf(std::ostream & os, Ind
 
   unsigned int i;
   os << indent << "Center: [";
-  for (i = 0; i < VImageDimension - 1; ++i)
+  for (i = 0; i + 1 < VImageDimension; ++i)
```

Comment:
We disallow Image's dimension to be zero.
```cpp
#include "itkImage.h"
int
main()
{
  auto zeroImage = itk::Image<short, 0>::New();
}
```
gives the following compile errors:
```text
1>------ Build started: Project: tester, Configuration: Debug x64 ------
1>tester.cpp
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkIndex.h(285,28): error C2338: Error: Only positive value sized VDimension allowed
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(133): message : see reference to class template instantiation 'itk::Index<0>' being compiled
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(89): message : see reference to class template instantiation 'itk::ImageBase<0>' being compiled
1>C:\Misc\Tester\tester.cpp(6): message : see reference to class template instantiation 'itk::Image<short,0>' being compiled
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkOffset.h(238,28): error C2338: Error: Only positive value sized VDimension allowed
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(138): message : see reference to class template instantiation 'itk::Offset<0>' being compiled
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkSize.h(226,28): error C2338: Error: Only positive value sized VDimension allowed
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(142): message : see reference to class template instantiation 'itk::Size<0>' being compiled
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkVector.h(63,1): error C2503: 'itk::FixedArray<T,0>': base classes cannot contain zero-sized arrays
1>        with
1>        [
1>            T=itk::ImageBase<0>::SpacingValueType
1>        ]
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(813): message : see reference to class template instantiation 'itk::Vector<itk::ImageBase<0>::SpacingValueType,0>' being compiled
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkPoint.h(54,1): error C2503: 'itk::FixedArray<T,0>': base classes cannot contain zero-sized arrays
1>        with
1>        [
1>            T=itk::ImageBase<0>::SpacingValueType
1>        ]
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(814): message : see reference to class template instantiation 'itk::Point<itk::ImageBase<0>::PointValueType,0>' being compiled
1>C:\Dev\ITK-git\Modules\ThirdParty\VNL\src\vxl\core\vnl\vnl_matrix_fixed.h(109,1): error C2087: 'data_': missing subscript
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkMatrix.h(297): message : see reference to class template instantiation 'vnl_matrix_fixed<T,0,0>' being compiled
1>        with
1>        [
1>            T=itk::SpacePrecisionType
1>        ]
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(815): message : see reference to class template instantiation 'itk::Matrix<itk::SpacePrecisionType,0,0>' being compiled
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(816,17): error C2229: class 'itk::ImageBase<0>' has an illegal zero-sized array
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(820,17): error C2229: class 'itk::ImageBase<0>' has an illegal zero-sized array
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(821,17): error C2229: class 'itk::ImageBase<0>' has an illegal zero-sized array
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(873,19): error C2229: class 'itk::ImageBase<0>' has an illegal zero-sized array
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageRegion.h(365,13): error C2229: class 'itk::ImageRegion<0>' has an illegal zero-sized array
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(875): message : see reference to class template instantiation 'itk::ImageRegion<0>' being compiled
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(876,14): error C2229: class 'itk::ImageBase<0>' has an illegal zero-sized array
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImageBase.h(877,14): error C2229: class 'itk::ImageBase<0>' has an illegal zero-sized array
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(89,1): error C2503: 'itk::ImageBase<0>': base classes cannot contain zero-sized arrays
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(131,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(134,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(135,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(138,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(141,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(142,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(148,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(152,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(156,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(157,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(161,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(168,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(392,3): error C3210: 'ImageBase<0>': a member using-declaration can only be applied to a base class member
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(101,3): error C3668: 'itk::Image<short,0>::CreateAnother': method with override specifier 'override' did not override any base class methods
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(104,3): error C3668: 'itk::Image<short,0>::GetNameOfClass': method with override specifier 'override' did not override any base class methods
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(190,3): error C3668: 'itk::Image<short,0>::Allocate': method with override specifier 'override' did not override any base class methods
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(195,3): error C3668: 'itk::Image<short,0>::Initialize': method with override specifier 'override' did not override any base class methods
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(381,3): error C3668: 'itk::Image<short,0>::Graft': method with override specifier 'override' did not override any base class methods
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(321,3): error C3668: 'itk::Image<short,0>::GetNumberOfComponentsPerPixel': method with override specifier 'override' did not override any base class methods
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(379,3): error C3668: 'itk::Image<short,0>::PrintSelf': method with override specifier 'override' did not override any base class methods
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(383,4): error C3668: 'itk::Image<short,0>::~Image': method with override specifier 'override' did not override any base class methods
1>C:\Dev\ITK-git\Modules\Core\Common\include\itkImage.h(391,3): error C3668: 'itk::Image<short,0>::ComputeIndexToPhysicalPointMatrices': method with override specifier 'override' did not override any base class methods
1>Done building project "tester.vcxproj" -- FAILED.
```

Reply:
Thank you for verifying that.